### PR TITLE
Add --sort and --reverse flags to list commands

### PR DESCRIPTION
**Pull request by jeremy**
## Summary

- Add client-side `--sort <field>` and `--reverse` flags to 7 list commands: todos, cards, messages, todolists, projects, people, and schedule entries
- Shared sort helpers in `sort.go` with per-type functions, field validation, and default sort directions (title/position ascending, created/updated descending, due ascending with empties last)
- Position sorting scoped to single-parent contexts only (single todolist or column); omitted from aggregate paths
- Todos aggregate path requires `--all` when `--sort` is passed (results are sampled per-list without it)
- People sorts raw `[]basecamp.Person` before slimming; uses `name` field (not `title`, which is job title)
- Existing hardcoded sorts preserved when `--sort` is not set, with one intentional change: people default sort is now case-insensitive (`strings.ToLower`) to match `--sort name` semantics and projects default sort

## Test plan

- [x] 42 unit tests in `sort_test.go` covering every sort function, field validation, due-date edge cases, reverse behavior across all types, and position scoping
- [x] `bin/ci` passes (formatting, vetting, linting, unit tests, e2e tests, surface snapshot, skill drift, bare groups)
- [x] Manual: `basecamp todos list --list <id> --sort due` shows soonest-first with empties at bottom
- [x] Manual: `basecamp projects list --sort created --reverse` shows oldest projects first
- [x] Manual: `basecamp todos list --sort title` (without --list or --all) returns usage error

### CODE REVIEW

File: internal/commands/cards.go [hunk 1]
```diff
@@ -153,6 +157,14 @@ func runCardsList(cmd *cobra.Command, project, column, cardTable string, limit,
 			return convertSDKError(err)
 		}
 
+		if sortField != "" {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/cards.go:168]: Fixed in 6405ffe — sort field now validated early, before any API calls. Also addressed --reverse on default sorts and schedule Summary fallback.

File: internal/commands/projects.go [hunk 2]
```diff
@@ -107,10 +111,16 @@ func runProjectsList(cmd *cobra.Command, status string, limit, page int, all boo
...
-	// Only sort when we have the full result set — alphabetizing a partial
-	// page would create a misleading view.
-	if page == 0 && limit == 0 {
+	if sortField != "" {
+		allowed := []string{"title", "created", "updated"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortProjects(projects, sortField, reverse)
+	} else if page == 0 && limit == 0 {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/projects.go:122]: Fixed in 6405ffe — sort field now validated early, before any API calls. Also addressed --reverse on default sorts and schedule Summary fallback.

File: internal/commands/people.go [hunk 3]
```diff
@@ -235,7 +239,20 @@ func runPeopleList(cmd *cobra.Command, projectID string, limit, page int, all bo
...
+	if sortField != "" {
+		allowed := []string{"name"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortPeople(people, sortField, reverse)
+	} else {
+		sort.Slice(people, func(i, j int) bool {
+			return people[i].Name < people[j].Name
+		})
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/people.go:254]: Fixed in 6405ffe — sort field now validated early, before any API calls. Also addressed --reverse on default sorts and schedule Summary fallback.

File: internal/commands/messages.go [hunk 4]
```diff
@@ -141,6 +145,14 @@ func runMessagesList(cmd *cobra.Command, project string, messageBoard string, li
 	}
 	messages := messagesResult.Messages
 
+	if sortField != "" {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/messages.go:154]: Fixed in 6405ffe — sort field now validated early, before any API calls. Also addressed --reverse on default sorts and schedule Summary fallback.

File: internal/commands/schedule.go [hunk 5]
```diff
@@ -210,6 +214,14 @@ func runScheduleEntries(cmd *cobra.Command, app *appctx.App, project, scheduleID
 	}
 	entries := entriesResult.Entries
 
+	if sortField != "" {
+		allowed := []string{"title", "created", "updated"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortScheduleEntries(entries, sortField, reverse)
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/schedule.go:224]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/people.go [hunk 6]
```diff
@@ -235,7 +239,20 @@ func runPeopleList(cmd *cobra.Command, projectID string, limit, page int, all bo
...
 
-	// Slim output and sort by name
+	// Sort raw people before slimming (sort functions need full SDK type)
+	if sortField != "" {
+		allowed := []string{"name"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortPeople(people, sortField, reverse)
+	} else {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/people.go:251]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/cards.go [hunk 7]
```diff
@@ -153,6 +157,14 @@ func runCardsList(cmd *cobra.Command, project, column, cardTable string, limit,
...
+			allowed := []string{"title", "created", "updated", "position", "due"}
+			if err := validateSortField(sortField, allowed); err != nil {
+				return err
+			}
+			sortCards(cardsResult.Cards, sortField, reverse)
+		}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/cards.go:170]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/cards.go [hunk 8]
```diff
@@ -203,6 +215,14 @@ func runCardsList(cmd *cobra.Command, project, column, cardTable string, limit,
...
 		}
 		allCards = cardsResult.Cards
+
+		if sortField != "" {
+			allowed := []string{"title", "created", "updated", "position", "due"}
+			if err := validateSortField(sortField, allowed); err != nil {
+				return err
+			}
+			sortCards(allCards, sortField, reverse)
+		}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/cards.go:225]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/todolists.go [hunk 9]
```diff
@@ -140,6 +144,14 @@ func runTodolistsList(cmd *cobra.Command, project, todosetFlag string, limit, pa
 	}
 	todolists := todolistsResult.Todolists
 
+	if sortField != "" {
+		allowed := []string{"title", "created", "updated", "position"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortTodolists(todolists, sortField, reverse)
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/todolists.go:154]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/todos.go [hunk 10]
```diff
@@ -405,6 +409,15 @@ func listTodosInList(cmd *cobra.Command, app *appctx.App, project, todolist, sta
...
 	todos := todosResult.Todos
 
+	// Apply client-side sort when requested
+	if sortField != "" {
+		allowed := []string{"title", "created", "updated", "position", "due"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortTodos(todos, sortField, reverse)
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/todos.go]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/todos.go [hunk 11]
```diff
@@ -518,6 +536,15 @@ func listAllTodos(cmd *cobra.Command, app *appctx.App, project, todosetFlag, ass
...
 	}
 
+	// Apply client-side sort when requested (--all already validated above)
+	if sortField != "" {
+		allowed := []string{"title", "created", "updated", "due"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortTodos(result, sortField, reverse)
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/todos.go]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/cards.go [hunk 12]
```diff
@@ -212,6 +232,15 @@ func runCardsList(cmd *cobra.Command, project, column, cardTable string, limit,
...
 		}
+
+		if sortField != "" {
+			// No position in aggregate — it's only meaningful within a single column
+			allowed := []string{"title", "created", "updated", "due"}
+			if err := validateSortField(sortField, allowed); err != nil {
+				return err
+			}
+			sortCards(allCards, sortField, reverse)
+		}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/cards.go:243]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/todos.go [hunk 13]
```diff
@@ -431,7 +444,12 @@ func listTodosInList(cmd *cobra.Command, app *appctx.App, project, todolist, sta
 	return app.OK(todos, respOpts...)
 }
 
-func listAllTodos(cmd *cobra.Command, app *appctx.App, project, todosetFlag, assignee, status string, overdue bool, limit int, all bool) error {
+func listAllTodos(cmd *cobra.Command, app *appctx.App, project, todosetFlag, assignee, status string, overdue bool, limit int, all bool, sortField string, reverse bool) error {
+	// Sorting the aggregate path without --all is misleading because results
+	// are silently sampled per-todolist using default SDK paging.
+	if sortField != "" && !all {
+		return output.ErrUsage("--sort requires --all when listing across todolists (results are sampled per list without it)")
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/todos.go:458]: The --sort requires --all guard is validated early in runTodosList now (before any API calls). The validation logic is tested via unit tests on validateSortField.

File: internal/commands/sort.go [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+		slices.Reverse(projects)
+	}
+}
+
+// sortPeople sorts a slice of people by name, then reverses if requested.
+func sortPeople(people []basecamp.Person, field string, reverse bool) {
+	sort.SliceStable(people, func(i, j int) bool {
+		switch field {
+		case "name":
+			return strings.ToLower(people[i].Name) < strings.ToLower(people[j].Name)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/sort.go:148]: Fixed — default people sort now uses strings.ToLower for consistency with --sort name.

File: internal/commands/projects.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/projects.go:122]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/messages.go [hunk 15]
```diff
@@ -141,6 +145,14 @@ func runMessagesList(cmd *cobra.Command, project string, messageBoard string, li
...
+		allowed := []string{"title", "created", "updated"}
+		if err := validateSortField(sortField, allowed); err != nil {
+			return err
+		}
+		sortMessages(messages, sortField, reverse)
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/messages.go:156]: Fixed in 6405ffe — validation moved before API calls.

File: internal/commands/sort.go [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+		slices.Reverse(people)
+	}
+}
+
+// sortScheduleEntries sorts a slice of schedule entries by field with default direction, then reverses if requested.
+func sortScheduleEntries(entries []basecamp.ScheduleEntry, field string, reverse bool) {
+	sort.SliceStable(entries, func(i, j int) bool {
+		switch field {
+		case "title":
+			return strings.ToLower(entries[i].Title) < strings.ToLower(entries[j].Title)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/sort.go]: Fixed in 6405ffe — sortScheduleEntries now uses Summary with Title as fallback via scheduleEntryTitle(), matching the display behavior in schedule show.

File: internal/commands/todos.go [hunk 17]
```diff
@@ -431,7 +445,15 @@ func listTodosInList(cmd *cobra.Command, app *appctx.App, project, todolist, sta
...
+	if sortField == "position" {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/todos.go]: Fixed in e6c9df2 — swapped the guard order so `--sort position` without `--list` now gets "requires --list" instead of the misleading --all message.

File: internal/commands/people.go [hunk 18]
```diff
@@ -235,7 +245,19 @@ func runPeopleList(cmd *cobra.Command, projectID string, limit, page int, all bo
...
+	// Sort raw people before slimming (sort functions need full SDK type)
+	if sortField != "" {
+		sortPeople(people, sortField, reverse)
+	} else {
+		sort.Slice(people, func(i, j int) bool {
+			return strings.ToLower(people[i].Name) < strings.ToLower(people[j].Name)
+		})
+		if reverse {
+			slices.Reverse(people)
+		}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/people.go:257]: Intentional change — the previous case-sensitive sort was inconsistent with projects (which already used `strings.ToLower`) and with `--sort name`. Updated PR description to call this out explicitly.

File: internal/commands/sort_test.go [hunk 19]
```diff
@@ -0,0 +1,346 @@
...
+
+func TestSortCards_Title(t *testing.T) {
+	cards := []basecamp.Card{
+		{Title: "Zebra"},
+		{Title: "Apple"},
+	}
+	sortCards(cards, "title", false)
+	assert.Equal(t, "Apple", cards[0].Title)
+	assert.Equal(t, "Zebra", cards[1].Title)
+}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/sort_test.go:141]: Added 10 more tests in ff31870 covering reverse and updated/created for cards, messages, todolists, and schedule entries. Now 42 tests total.
